pennylane/io.py

```diff
@@ -122,11 +109,177 @@ def from_qiskit(quantum_circuit, measurements=None):
+    which device wires it should operate on. If no wires are passed, it will default
+    to sequential wire labels starting at 0.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit():
+            my_qfunc(wires=["a", "b"])
+            return qml.expval(qml.Z("a")), qml.var(qml.Z("b"))
+
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    The measurements can also be passed directly to the function when creating the
+    qfunc, making it possible to create a PennyLane circuit with ``qml.QNode`:
+
+    >>> measurements = [qml.expval(qml.Z(0)), qml.var(qml.Z(1))]
+    >>> circuit = qml.QNode(qml.from_qiskit(qc, measurements), dev)
+    >>> circuit()
+    (tensor(0.70738827, requires_grad=True),
+    tensor(0.99999937, requires_grad=True))
+
+    .. note::
 
         The ``measurement`` keyword allows one to add a list of PennyLane measurements
-        that will override the terminal measurements present in their ``QuantumCircuit``.
+        that will **override** the terminal measurements present in the ``QuantumCircuit``,
+        so that they are not performed before the operations specified in ``measurements``.
+        Converting a QuantumCircuit with ``measurements`` set will create a quanutm function
+        that does not return final or mid-circuit measurement values. See Usage Details below
+        for more information on how measurements defined on the QuantumCircuit are returned if
+        ``measurements=None``.
+
+    If an existing ``QuantumCircuit`` already ends with final measurements, the QNode can return
+    the expectation values of those final measurements directly:
+
+     .. code-block:: python
+
+        qc = QuantumCircuit(2, 2)
+        qc.rx(np.pi, 0)
+        qc.measure_all()
+
+        @qml.qnode(dev)
+        def circuit():
+            # here measurements=None, so the measurements present in the QuantumCircuit are returned
+            measurements = qml.from_qiskit(qc)()
+            return [qml.expval(m) for m in measurements]
+
+    >>> circuit()
+    [tensor(1., requires_grad=True), tensor(0., requires_grad=True)]
+
+    .. note::
+
+        The returned measurements from the ``QuantumCircuit`` are in the measurement basis
+        and are readout values between 0 (ground) and 1 (excited). Such a measurement on the
+        `ith` wire does not match the result of `qml.expval(qml.PauliZ(i)`, which ranges from
+        1 (ground) to -1 (excited), indicating a position on the Bloch sphere.
+
+        I.e. we can alternatively translate the circuit and capture the same information in
+        a different format as:
+
+        >>> measurements = [qml.expval(qml.Z(0)), qml.expval(qml.Z(1))]
+        >>> circuit = qml.QNode(qml.from_qiskit(qc, measurements), dev)
+        >>> circuit()
+        [tensor(-1., requires_grad=True), tensor(1., requires_grad=True)]
```

Comment:
```suggestion
        The `measurements` returned from a ``QuantumCircuit`` are in the computational basis
        with `0` corresponding to :math:`|0\rangle` and `1` corresponding to :math:`|1\rangle`.
```
Happy to discuss, but I felt like a simple statement was easier to understand here.

Reply:
@lillian542 Is there a good pedagogical reason why we draw a comparison to the Bloch sphere here?

Reply:
No, it doesn't actually explain anything, you can absolutely just label your Bloch sphere 0 and 1 instead, so I'd say it adds nothing. 

But I think its worth considering commenting on the discrepancy between what will be output if you take `expval(m)` for a measurement from the Qiskit circuit versus if you do `expval(qml.Z(0))`. We have a tendency to talk about `expval(qml.Z(0))` as being "in the measurement basis", but mapping ground and excited to 0 and 1 is quite different than mapping it to 1 and -1, especially in terms of how you interpret the number 1.

On the other hand, maybe most people are not thinking about that and will not be confused, and it's better to keep it simple.

Reply:
I think we can mention the distinction briefly:

```
        The `measurements` returned from a ``QuantumCircuit`` are in the computational basis
        with `0` corresponding to :math:`|0\rangle` and `1` corresponding to :math:`|1\rangle`.
        This corresponds to the :math:`|1 \rangle \langle 1|` observable rather than the
        :math:`Z` Pauli operator. 
```